data/pools/transactionPool.go

```diff
@@ -172,6 +174,11 @@ func (pool *TransactionPool) Test(txgroup []transactions.SignedTxn) error {
 		txgroup[i].InitCaches()
 	}
 
+	pendingSize := len(pool.Pending())
```

Comment:
Nit: Do we want to move this into `test`? That way `Remember` also picks up this behavior.

Reply:
I'm not strongly opinionated around that. If someone places too much load on his own node, that's his issue.
I'll PR that tonight and we can discuss this issue separately ( once it's not on the critical path )